src/Core/Tools/Models/Data/SendFileData.cs

```diff
@@ -1,22 +1,54 @@
-﻿using System.Text.Json.Serialization;
+﻿#nullable enable
+
+using System.Text.Json.Serialization;
+using static System.Text.Json.Serialization.JsonNumberHandling;
 
 namespace Bit.Core.Tools.Models.Data;
 
+/// <summary>
+/// A file secret being sent.
+/// </summary>
 public class SendFileData : SendData
 {
+    /// <summary>
+    /// Instantiates a <see cref="SendFileData"/>.
+    /// </summary>
     public SendFileData() { }
 
-    public SendFileData(string name, string notes, string fileName)
+    /// <inheritdoc cref="SendFileData()"/>
+    /// <param name="name">Attached file name.</param>
+    /// <param name="notes">User-provided private notes of the send.</param>
+    /// <param name="fileName">Attached file name.</param>
+    public SendFileData(string name, string? notes, string fileName)
         : base(name, notes)
     {
         FileName = fileName;
     }
 
-    // We serialize Size as a string since JSON (or Javascript) doesn't support full precision for long numbers
-    [JsonNumberHandling(JsonNumberHandling.WriteAsString | JsonNumberHandling.AllowReadingFromString)]
+    /// <summary>
+    /// Size of the attached file in bytes.
+    /// </summary>
+    /// <remarks>
+    /// Serialized as a string since JSON (or Javascript)  doesn't support 
+    /// full precision for long numbers
+    /// </remarks>
+    [JsonNumberHandling(WriteAsString | AllowReadingFromString)]
     public long Size { get; set; }
 
-    public string Id { get; set; }
-    public string FileName { get; set; }
+    /// <summary>
+    /// Uniquely identifies the file.
+    /// </summary>
+    public string? Id { get; set; }
```

Comment:
I like the extra comments you have about this in the PR description, mind adding them here as well? Also, should this maybe be marked with `[DisallowNull]` while it being nullable is the right annotation I'm guessing we should never assign it to a non-null ID.

Reply:
Knowing why the field is null and what it is useful for indicating seems valuable in a comment without crossing domain boundaries. Maybe something like: `Expected to be <see langword="null" /> when the file has not been uploaded yet, and set to a unique identifier for a file provider once the file has been uploaded.`?

Reply:
I've included guidance about its nullability in the latest commit. I also updated the file name member to distinguish when it should/should not contain an empty string value.